### ISSUE TITLE
Support definite assignment in TypesScript 2.7

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3679]
Starting TypeScript 2.7 there is a special syntax for definite assignment.
https://blogs.msdn.microsoft.com/typescript/2018/01/17/announcing-typescript-2-7-rc/

TSLint should recognize this pattern and not issue error of noUninitialized.
In other words this should trigger this automatically: // tslint:disable-next-line no-uninitialized

```ts
class C {
    foo!: number;
}
```

**Comment by ajafff** [comment]
> no-uninitialized is not a TSLint core rule. You should bug the author of tslint-strict-null-checks

**Comment by NN---** [comment]
Thanks for quick reply.
Any reason to not put it as TSLint core rule ?

**Comment by ajafff** [comment]
Now that the compiler has this builtin check, I don't think we need to duplicate that implementation.
Also this would need some new APIs to do this as reliable as the compiler does.
